products/monitoring/api.yaml

```diff
@@ -964,6 +962,141 @@ objects:
+      latency below 200 milliseconds" or "99.5% of requests in each calendar
+      month return successfully."
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Service Monitoring': 'https://cloud.google.com/monitoring/service-monitoring'
+        'Monitoring API Documentation': 'https://cloud.google.com/monitoring/api/v3/'
+      api: 'https://cloud.google.com/monitoring/api/ref_v3/rest/v3/services.serviceLevelObjectives'
+    parameters:
+    - !ruby/object:Api::Type::String
+      name: service
+      required: true
+      url_param_only: true
+      input: true
+      description: |
+        ID of the service to which this SLO belongs.
+    - !ruby/object:Api::Type::String
+      name: sloId
+      description: |
+        The id to use for this ServiceLevelObjective. If omitted, an id will be generated instead.
+      input: true
+    properties:
+    - !ruby/object:Api::Type::String
+      name: name
+      description: |
+        The full resource name for this service. The syntax is:
+        projects/[PROJECT_ID_OR_NUMBER]/services/[SERVICE_ID]/serviceLevelObjectives/[SLO_NAME]
+      output: true
+    - !ruby/object:Api::Type::String
+      name: displayName
+      description: |
+        Name used for UI elements listing this SLO.
+    - !ruby/object:Api::Type::Double
+      name: goal
+      required: true
+      description: |
+        The fraction of service that must be good in order for this objective
+        to be met. 0 < goal <= 0.999
+    - !ruby/object:Api::Type::String
+      name: rollingPeriod
+      exactly_one_of:
+        - rolling_period
+        - calendar_period
+      description: |
+        A rolling time period, semantically "in the past <rollingPeriod>".
+        Must be an integer multiple of 1 day no larger than 30 days.
+    - !ruby/object:Api::Type::Enum
+      name: calendarPeriod
+      exactly_one_of:
+        - rolling_period
+        - calendar_period
+      description: |
+        A calendar period, semantically "since the start of the current
+        <calendarPeriod>".
+      values:
+        - DAY
+        - WEEK
+        - FORTNIGHT
+        - MONTH
+    - !ruby/object:Api::Type::NestedObject
+      name: serviceLevelIndicator
+      description: |
+        serviceLevelIndicator (SLI) describes a good service.
+        It is used to measure and calculate the quality of the Service's
+        performance with respect to a single aspect of service quality.
+      properties:
+        - !ruby/object:Api::Type::NestedObject
+          name: basicSli
+          exactly_one_of:
+            - basic_sli
+          description: |
+            Basic Service-Level Indicator (SLI) on a well-known service type.
+            Performance will be computed on the basis of pre-defined metrics.
+
+            SLIs are used to measure and calculate the quality of the Service's
+            performance with respect to a single aspect of service quality.
+          properties:
```

Comment:
Right now all the fields in this nested object are optional. Can we use one of the helpers to make sure people don't set an empty block?

Reply:
is that covered by 
```
  exactly_one_of:
    - basic_sli.0.latency
```
?

Reply:
Same as the other one, it ends up as an empty list. Have you tried not setting that field to see whether the empty list works to set it as required?

Reply:
resolved by setting basic_sli to required